Assets/Scripts/SpotArea.cs

```diff
@@ -0,0 +1,142 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpotArea : MonoBehaviour
+{
+    const int countTime = 3;
+    const float SPOT_ANGLE = 12.0f;
+
+    [SerializeField]
+    private string targetTag;
+
+    [SerializeField]
+    private UnityChanController unityChan;
+
+    int counter = 0;
+    Coroutine timerCoroutine;
+    GameManager gameManager;
+    SpotCreator spotCreator;
+    Light spotLight;
+    private AudioSource[] spotSE;
+    private StageManager stageManager;
+    private bool spotJudge;
+
+    void Start()
+    {
+        // GameManager�C���X�^���X�擾
+        gameManager = GameObject.Find("GameManager").GetComponent<GameManager>();
+        stageManager = GameObject.Find("StageManager").GetComponent<StageManager>();
+        spotSE = GetComponents<AudioSource>();
+
+        // SpotCreator�C���X�^���X�擾
+        spotCreator = GameObject.Find("SpotLight").GetComponent<SpotCreator>();
+        spotLight = spotCreator.GetComponent<Light>();
+
+        spotJudge = false;
+    }
+    
+    void Update()
+    {
+        if (gameManager.GameOverFlg)
+        {
+            // �R���[�`���������Ă���ꍇ
+            if (null != timerCoroutine)
+            {
+                StopCoroutine(timerCoroutine);
+                spotSE[0].Stop();
+                //gameManager.GameClearFlg = false;
+                gameManager.GameOverFlgSet(false);
+            }
+        }
+    }
+    
+    private void OnTriggerStay(Collider other)
+    {
+        // �Q�[���I�[�o�[�̏ꍇ�̓N���A���f�����͎�s�����Ȃ�
+        if (gameManager.GameOverFlg)
+        {
+            //gameOverFlg = true;
+            // �R���[�`���������Ă���ꍇ
+            if (null != timerCoroutine)
+            {
+                StopCoroutine(timerCoroutine);
+            }
+            // �G���A��ɗ��܂�Ă���Ƃ���SE��~�߂�
+            spotSE[0].Stop();
+            return;
+        }
+
+        // ���p�x�����ɂȂ����Փ˔���͂��Ȃ�
+        if (spotLight.spotAngle < SPOT_ANGLE)
+        {
+            // �R���[�`���������Ă���ꍇ
+            if (null != timerCoroutine)
+            {
+                StopCoroutine(timerCoroutine);
+            }
+
+            // �G���A��ɗ��܂�Ă���Ƃ���SE��~�߂�
+            spotSE[0].Stop();
+            return;
+        }
+
+        // �Q�[���I�[�o�[�ƂȂ�Ă��Ȃ��ꍇ ���A1�x��G���A���菈����s��Ă��Ȃ��ꍇ
```

Comment:
ロジック的にはこれですか？ (エリア => クリア)

```suggestion
        // ゲームオーバーとなっていない場合 かつ、1度もクリア判定処理を行っていない場合
```